Scripts/package.py

```diff
@@ -76,50 +68,47 @@ def flush(*args, **kwargs):
     sys.stdout.flush()
 
 class Release:
-    @staticmethod
-    def parse_zip_name(name):
-        m = Z3_RELEASE_REGEXP.match(name)
-        if not m:
-            raise Exception("{} does not match Z3_RELEASE_REGEXP".format(name))
-        return m.group('platform'), m.group('os'), m.group("directory")
-
-    def __init__(self, js, version, out):
-        self.z3_name = js["name"]
-        self.size = js["size"]
-        self.url = js["browser_download_url"]
-        self.platform, self.os, self.directory = Release.parse_zip_name(js["name"])
+
+    def __init__(self, os, platform, version, out):
+        self.z3_zips = [ "z3-{}-{}-bin.zip".format(z3_version, os) for z3_version in Z3_VERSIONS ]
+        self.platform, self.os = platform, os
         self.os_name = self.os.split("-")[0]
-        self.z3_zip = path.join(CACHE_DIRECTORY, self.z3_name)
         self.dafny_name = "dafny-{}-{}-{}.zip".format(version, self.platform, self.os)
         if out != None:
             self.dafny_name = out
-        self.target = "{}-{}".format(z3ToDotNetOSMapping[self.os_name], self.platform)
+        self.target = "{}-{}".format(gitHubToDotNetOSMapping[self.os_name], self.platform)
         self.dafny_zip = path.join(DESTINATION_DIRECTORY, self.dafny_name)
         self.buildDirectory = path.join(BINARIES_DIRECTORY, self.target, "publish")
 
-    @property
-    def cached(self):
-        return path.exists(self.z3_zip) and path.getsize(self.z3_zip) == self.size
+    def url(self, z3_zip):
+        return "{}/{}".format(Z3_URL_BASE, z3_zip)
+
+    def local_zip(self, z3_zip):
+        return path.join(CACHE_DIRECTORY, z3_zip)
 
     @property
-    def MB(self):
-        return self.size / 1e6
+    def cached(self):
+        for z3_zip in self.z3_zips:
+          if not path.exists(self.local_zip(z3_zip)):
+              return False
+        return True
```

Comment:
nit: can we rename these to clarify that they're referring to Z3 and not Dafny or something else?

Reply:
Good call. I'll do that.